### PR TITLE
[IOTDB-5921] Pipe: config & descriptor

### PR DESCRIPTION
**Pull request by MiniSho**

### CODE REVIEW

File: node-commons/src/assembly/resources/conf/iotdb-common.properties [hunk 1]
```diff
@@ -989,6 +989,46 @@ cluster_name=defaultCluster
...
+# pipe_initial_sync_delay_minutes = 3
+
+# The sync regular interval (in minutes)for the PipeMetaSyncer service.
+# pipe_sync_interval_minutes = 3
+
+# The delay period (in milliseconds) between each retry when a connection failure occurs.
+# pipe_retry_interval_ms = 1000
+
+# The size of the pending queue for the PipeConnector to store the events.
+# pipe_connector_pending_queue_size = 65536
```
Comment (SteveYurongSu) [inline node-commons/src/assembly/resources/conf/iotdb-common.properties:1024]:
```suggestion
# pipe_connector_pending_queue_size = 1024
```

File: node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java [hunk 2]
```diff
@@ -130,6 +130,44 @@ public class CommonConfig {
...
+  private int connectorPendingQueueSize = 1024;
+
+  private int basicCheckPointIntervalByConsumedEventCount = 10_000;
+
+  private long basicCheckPointIntervalByTimeDuration = 10 * 1000L;
+
+  private long pendingQueueMaxBlockingTimeMs = 1000;
+
+  /** The maximum number of threads that can be used to execute subtasks in PipeSubtaskExecutor */
+  private int pipeMaxThreadNum = 5;
```
Comment (SteveYurongSu) [inline node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java:169]: ?

Comment (SteveYurongSu) [reply node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java:169]: node-commons/src/assembly/resources/conf/iotdb-common.properties

File: node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonDescriptor.java [hunk 3]
```diff
@@ -204,6 +206,89 @@ public void loadCommonProps(Properties properties) {
...
+                "pipe_basic_check_point_interval_by_consumed_event_count",
+                String.valueOf(config.getBasicCheckPointIntervalByConsumedEventCount()))));
+
+    config.setBasicCheckPointIntervalByTimeDuration(
+        Integer.parseInt(
+            properties.getProperty(
+                "pipe_basic_check_point_interval_by_time_duration",
+                String.valueOf(config.getBasicCheckPointIntervalByTimeDuration()))));
+  }
+
```
Comment (SteveYurongSu) [inline node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonDescriptor.java:291]: pendingQueueMaxBlockingTimeMs

**Review by SteveYurongSu — CHANGES_REQUESTED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Review by MiniSho — COMMENTED** [review]
LGTM

**Pull request merged**